file_system.go

```diff
@@ -420,7 +508,7 @@ func parseMLST(entry string, skipSelfParent bool) (os.FileInfo, error) {
 		return nil, incompleteError
 	}
 
-	if skipSelfParent && (typ == "cdir" || typ == "pdir" || typ == "." || typ == "..") {
+	if skipSelfParent && (typ == "cdir" || typ == "pdir" || typ == "." || typ == ".."|| parts[1] == "." || parts[1] == "..") {
```

Comment:
No sane server should return "." or ".." as type "dir"...
I noticed this as my walk function entered a infinite loop, tcpdump helped identify the problem.

How would you test for this?


How